front/components/Layout/Header/RightNav/UserCard/UserCard.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { MockedProvider } from '@apollo/client/testing';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import Home from './index';
+
+// eslint-disable-next-line
+const useRouter = jest.spyOn(require('next/router'), 'useRouter');
+// eslint-disable-next-line
+const useViewer = jest.spyOn(require('hooks/auth'), 'useViewer');
+
+test('UserCard expandLess', () => {
+  const mocks = [];
+
+  useRouter.mockImplementation(() => ({
+    route: '',
+    pathname: '',
+    query: '',
+    asPath: '',
+  }));
+
+  useViewer.mockImplementation(() => ({
+    username: 'toto',
+    email: 'tata',
+  }));
```

Comment:
I think it would be better if we could intercept the requests (using [`msw`](https://mswjs.io/) for example) instead of mocking our hooks. It would give us way more confidence.

Reply:
I use the msw of appollo to make it (https://www.apollographql.com/docs/react/development-testing/testing/)